lib/active_merchant/billing/gateways/omise.rb

```diff
@@ -0,0 +1,376 @@
+
+      # Credit cards supported by Omise
+      # * VISA
+      # * MasterCard
+      self.supported_cardtypes = [:visa, :master]
+
+      # Omise main page
+      self.homepage_url = 'https://www.omise.co/'
+      self.display_name = 'Omise'
+
+      # Creates a new OmiseGateway.
+      #
+      # Omise requires public_key for token creation.
+      # And it requires secret_key for other transactions.
+      # These keys can be found in https://dashboard.omise.co/test/api-keys
+      #
+      # ==== Options
+      #
+      # * <tt>:public_key</tt> -- Omise's public key (REQUIRED).
+      # * <tt>:secret_key</tt> -- Omise's secret key (REQUIRED).
+
+      def initialize(options={})
+        requires!(options, :public_key, :secret_key)
+        @public_key = options[:public_key]
+        @secret_key = options[:secret_key]
+        super
+      end
+
+      # Perform a purchase.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>   -- The purchasing amount as the value is in Satang.
+      # * <tt>payment</tt> -- The CreditCard object or card token that is used for the transaction.
+      # * <tt>options</tt> -- An optional parameters.
+      #
+      # ==== Options
+      # * <tt>token_id</tt> -- token id, use Omise.js library to retrieve a token id
+      # if this is passed as an option, it will ignore tokenizing via Omisevaultgateway object
+      #
+      # === Example
+      #  To create a charge on a card
+      #
+      #   purchase(money, creditcard_object)
+      #
+      #  To create a charge on a token
+      #
+      #   purchase(money, nil, { :token_id => token_id, ... })
+      #
+      #  To create a charge on a customer
+      #
+      #   purchase(money, nil, { :customer_id => customer_id })
+
+      def purchase(money, payment, options={})
+        create_post_for_purchase(money, payment, options)
+      end
+
+      # Authorize a charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>   -- An amount of money to charge in Satang.
+      # * <tt>payment</tt> -- The CreditCard
+      # * <tt>options</tt> -- A standard hash options
+
+      def authorize(money, payment, options={})
+        options[:capture] = 'false'
+        create_post_for_purchase(money, payment, options)
+      end
+
+      # Capture or a pre-authorized charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>money</tt>     -- An amount of money to charge in Satang.
+      # * <tt>charge_id</tt> -- The Charge object identifier.
+      # * <tt>options</tt>   -- A standard hash options.
+
+      def capture(money, charge_id, options={})
+        post = {}
+        add_amount(post, money, options)
+        commit(:post, "charges/#{CGI.escape(charge_id)}/capture", post, options)
+      end
+
+      # Void a charge.
+      #
+      # ==== Parameters
+      #
+      # * <tt>charge_id</tt> -- The Charge object identifier.
+      # * <tt>options</tt>   -- A standard hash options.
+
+      def void(charge_id, options={})
```

Comment:
Void is used by some banks to refer to the cancelling of a capture before settlement. The bank we're using calls it "reverse" to cancel an authorization.

But anyhow we currently don't support either of them because of the added complexity this feature added to our charge object representation. Furthermore since the issuing bank will reverse the auth by themselves after some time we decided to keep go without those.